simplejson/simplejson.go

```diff
@@ -0,0 +1,268 @@
+package simplejson
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"reflect"
```

Comment:
IMO too much use of reflection. Can be done without it.

Reply:
I'm using reflect when this might help with performance (passing reflect.Value instead of "converting" a slice of a certain type to interface slice) but also to add some generic implementation, knowing that GO is not quite generic friendly. The latter is used only to set the struct attributes and can be quickly replaced with switch/case.

Reply:
Have a look at https://github.com/mitchellh/mapstructure (we use it in another project). I think it's better to have parsers return a map and then use `mapstructure` to populate the structure.

Reply:
will check it out, thanks